lib/virt_autotest/utils.pm

```diff
@@ -85,19 +104,45 @@ sub restart_modular_libvirt_daemons {
     } else {
         # Restart the sockets first
         foreach my $daemon (@daemons) {
-            systemctl("restart $daemon\{,-ro,-admin\}.socket");
+            if ($daemon == 'lock') {
+                systemctl("restart virt${daemon}d\{,-admin\}.socket");
+            } else {
+                systemctl("restart virt${daemon}d\{,-ro,-admin\}.socket");
+            }
         }
 
         # Introduce idle time here (e.g., sleep 5) if necessary
         sleep 5;
 
         # Restart the services after a brief idle time
         foreach my $daemon (@daemons) {
-            systemctl("restart $daemon.service");
+            systemctl("restart virt${daemon}d.service");
         }
     }
 
-    record_info("Libvirt daemons restarted", join(' ', @daemons));
+    record_info("Modular Libvirt daemons restarted", join(' ', @daemons));
+    check_modular_libvirt_daemons;
+}
+
+# Restart monolithic libvirt
+sub restart_libvirtd {
+    if (is_sle('<12')) {
+        assert_script_run('rclibvirtd restart', 180);
+    }
+    elsif (is_alp) {
+        my $_libvirtd_pid = script_output(q@ps -ef |grep [l]ibvirtd | gawk '{print $2;}'@);
+        my $_libvirtd_cmd = script_output("ps -o command $_libvirtd_pid | tail -1");
+        assert_script_run("kill -9 $_libvirtd_pid");
+        assert_script_run("$_libvirtd_cmd");
+    }
+    elsif (is_modular_libvirtd) {
+        restart_modular_libvirt_daemons;
+    }
+    else {
+        systemctl("restart libvirtd", timeout => 180);
+    }
+    record_info("Libvirtd Daemon has been restarted!");
+    save_screenshot;
```

Comment:
What about sle-micro ? Do you want to cover it in ```restart_libvirtd``` ?

Reply:
I was not clear if setup modular libvirt as the default in sle-micro

Reply:
Short update,  adapt SLE automation both Monolithic and Modular libvirt together for our SLE KVM and XEN vm host as this PR purpose.  regarding our sle micro with the modular libvirt, we need do more investigation and create a new PR to follow it.